docs/source/_templates/sdk_TP_tutorial.rst

```diff
@@ -109,53 +109,68 @@ and transaction family specific. When implementing a transaction handler the
 binary serialization protocol is up to the implementer.
 {% endif %}
 
-Without yet getting into the details of how this information is encoded, we can
-start to think about what ``apply`` needs to do. ``apply`` needs to
+To separate details of state encoding and payload handling from validation
+logic, the Xo example has XoState and XoPayload classes. The XoPayload has
+name, action, and space fields, while the XoState contains information about
+the game name, board, state, and which players are playing in the game.
 
-1) unpack the command data from the transaction,
-2) retrieve the game data from the context,
-3) play the game, and
-4) save the updated game data.
-
-Accordingly, a top-down approach to ``apply`` might look like this:
+The actions one can enact on a game are to create the game, take a new board space,
+and delete the game.
 
 {% if language == 'JavaScript' %}
 
 .. code-block:: javascript
 
-      apply (transactionProcessRequest, stateStore) {
-        return _unpackTransaction(transactionProcessRequest)
-        .then((transactionData) => {
-
-        let stateData = _getStateData(transactionData.gameName, stateStore)
-
-        let updatedGameData = _playXO(
-          stateData.board,
-          stateData.state,
-          stateData.player1,
-          stateData.player2,
-          transactionData.signer,
-          transactionData.action,
-          transactionData.space
-        )
-        _storeGameData(transactionData.gameName, updatedGameData, stateStore)
-        })
-      }
+    apply (transactionProcessRequest, context) {
+        let payload = XoPayload.fromBytes(transactionProcessRequest.payload)
+        let xoState = new XoState(context)
+        let header = transactionProcessRequest.header
+        let player = header.signerPublicKey
+        if (payload.action === 'create') {
+            ...
+        } else if (payload.action === 'take') {
+            ...
+        } else if (payload.action === 'delete') {
+            ...
+        } else {
+            throw new InvalidTransaction(
+                'Action must be create or take not ${payload.action}'
```

Comment:
Missing delete in the InvalidTransaction message. This needs to be updated in the code as well.